### PR TITLE
💥 enable network errors removal

### PR DESCRIPTION
**Pull request by amortemousque**
## Motivation

RUM errors are not the best way to trigger attention for network requests with a bad status code. We already collect that information within RUM resources. By stopping the generation of RUM errors for these resources, we simplify the value proposition of RUM errors.

## Changes

- Remove the feature flag
- Move network error code in Logs

## Testing

Unit, Locally

---

I have gone over the [contributing](https://github.com/DataDog/browser-sdk/blob/main/CONTRIBUTING.md) documentation.

### CODE REVIEW

File: test/e2e/scenario/rum/errors.scenario.ts [hunk 1]
```diff
@@ -18,59 +17,4 @@ describe('rum errors', () => {
         expect(browserLogs.length).toEqual(1)
       })
     })
-
-  createTest('send XHR network errors')
```
Comment (BenoitZugmeyer) [inline test/e2e/scenario/rum/errors.scenario.ts:22]: Maybe those tests could be moved to `logs.scenario.ts`?

**Review by BenoitZugmeyer — APPROVED** [review]

File: packages/logs/src/domain/trackNetworkError.ts [hunk 2]
```diff
@@ -12,20 +12,12 @@ import {
...
-  trackAbortedRequests = true
-) {
+export function trackNetworkError(configuration: Configuration, errorObservable: Observable<RawError>) {
   startXhrProxy().onRequestComplete((context) => handleCompleteRequest(RequestType.XHR, context))
   startFetchProxy().onRequestComplete((context) => handleCompleteRequest(RequestType.FETCH, context))
 
   function handleCompleteRequest(type: RequestType, request: XhrCompleteContext | FetchCompleteContext) {
-    if (
-      !configuration.isIntakeUrl(request.url) &&
-      (trackAbortedRequests || !request.isAborted) &&
```
[the remark this answers is not in the document]
Comment (BenoitZugmeyer) [reply packages/logs/src/domain/trackNetworkError.ts:19]: > to keep it for now
Just to be sure, are you talking about the `remove-network-error` flag?

[the remark this answers is not in the document]
Comment (bcaudan) [reply packages/logs/src/domain/trackNetworkError.ts:19]: I am concerned about the logs errors for request aborted programatically.

[the remark this answers is not in the document]
Comment (bcaudan) [reply packages/logs/src/domain/trackNetworkError.ts:19]: I got confused, all good 👍

**Review by bcaudan — APPROVED** [review]

**Pull request merged**